records_mover/records/targets/table/move_from_dataframes_source.py

```diff
@@ -30,25 +30,27 @@ def __init__(self,
         super().__init__(prep, target_table_details, processing_instructions)
 
     def move(self) -> MoveResult:
-        if len(self.table_target.known_supported_records_formats()) != 0:
-            if self.table_target.can_move_from_fileobjs_source():
-                return self.move_from_dataframes_source_via_fileobjs()
-            else:
-                # Some databases, like Redshift, can't load from a
-                # stream, but can load from files on an object store
-                # they're pointed to.
-                return self.move_from_dataframes_source_via_records_directory()
+        target_supports_formats = len(self.table_target.known_supported_records_formats()) != 0
+        if (target_supports_formats and self.table_target.can_move_from_fileobjs_source()):
+            return self.move_from_dataframes_source_via_fileobjs()
+        elif (target_supports_formats and
+              self.table_target.can_move_from_temp_loc_after_filling_it()):
+            # Some databases, like Redshift, can't load from a
+            # stream, but can load from files on an object store
+            # they're pointed to.
+            return self.move_from_dataframes_source_via_temporary_records_directory()
         else:
             logger.info("Known formats for target database: "
                         f"{self.table_target.known_supported_records_formats()}")
             logger.info("Table target can move from fileobjs source? "
                         f"{self.table_target.can_move_from_fileobjs_source()}")
             logger.warning("Loading via INSERT statement as this DB "
-                           "driver does not yet support more direct LOAD methods.  "
+                           "driver does not yet support or is not configured for "
+                           "more direct load methods.  "
                            "This may be very slow.")
             return self.move_from_dataframes_source_via_insert()
 
-    def move_from_dataframes_source_via_records_directory(self) -> MoveResult:
+    def move_from_dataframes_source_via_temporary_records_directory(self) -> MoveResult:
```

Comment:
Clarify that this function uses a temp directory, as it was a bit of a surprise to me as well by the original name.